core/src/main/scala/kafka/consumer/BaseConsumer.scala

```diff
@@ -45,21 +57,34 @@ case class BaseConsumerRecord(topic: String,
                               key: Array[Byte],
                               value: Array[Byte])
 
-class NewShinyConsumer(topic: Option[String], whitelist: Option[String], consumerProps: Properties, val timeoutMs: Long = Long.MaxValue) extends BaseConsumer {
+class NewShinyConsumer(topic: Option[String], partitionId: Option[Int], offset: Option[Long], whitelist: Option[String], consumerProps: Properties, val timeoutMs: Long = Long.MaxValue) extends BaseConsumer {
   import org.apache.kafka.clients.consumer.KafkaConsumer
 
   import scala.collection.JavaConversions._
 
   val consumer = new KafkaConsumer[Array[Byte], Array[Byte]](consumerProps)
-  if (topic.isDefined)
-    consumer.subscribe(List(topic.get))
-  else if (whitelist.isDefined)
-    consumer.subscribe(Pattern.compile(whitelist.get), new NoOpConsumerRebalanceListener())
-  else
-    throw new IllegalArgumentException("Exactly one of topic or whitelist has to be provided.")
-
+  consumerInit()
   var recordIter = consumer.poll(0).iterator
 
+  def consumerInit() {
+    if (topic.isDefined)
+      if (partitionId.isDefined) {
+        val topicPartition = new TopicPartition(topic.get, partitionId.get)
+        consumer.assign(List(topicPartition))
+        offset.get match {
+          case OffsetRequest.EarliestTime => consumer.seekToBeginning(List(topicPartition))
+          case OffsetRequest.LatestTime => consumer.seekToEnd(List(topicPartition))
+          case _ => consumer.seek(topicPartition, offset.get)
+        }
+      }
+      else
+        consumer.subscribe(List(topic.get))
+    else if (whitelist.isDefined)
+      consumer.subscribe(Pattern.compile(whitelist.get), new NoOpConsumerRebalanceListener())
+    else
+      throw new IllegalArgumentException("Exactly one of topic or whitelist has to be provided.")
+  }
```

Comment:
@ijuma Regarding the exception, I don't think the control would ever reach that last `case` because arguments are checked before that [here](https://github.com/vahidhashemian/kafka/blob/5058e4f1c81ed8f47f021248490cf8d3ad1ecc3c/core/src/main/scala/kafka/tools/ConsoleConsumer.scala#L321). I can update the message to something more generic like `Invalid combination of arguments is provided.` What do you think?